src/Test.py

```diff
@@ -105,6 +108,44 @@ def test_quorum_of_nodes(self):
             log.test.debug('Node %s, quorum_set = %s',node.name,node.quorum_set)
             log.test.debug('Node %s, check_threshold = %s',node.name,node.quorum_set.get_quorum())
 
+    def test_prepare_nomination_msg_correctly_adds_value(self):
+        for topology in ['FULL','ER']:
+            nodes = Network.generate_nodes(n_nodes=5, topology=topology)
+            mempool = Mempool()
+            for node in nodes:
+                node.attach_mempool(mempool)
+
+            for i in range(5):
+                mempool.mine()
+
+            for node in nodes:
+                node.prepare_nomination_msg()
+
+            for node in nodes:
+                self.assertEqual(type(node.nomination_state['voted'][0]), Value)
+                # as extend is used to add new Values the list of
+                self.assertTrue(len(node.nomination_state['voted'][0].transactions) == len(node.ledger.transactions))
+                self.assertTrue(len(node.storage.messages) > 0)
+
+    def test_prepare_nomination_phase_correctly_adds_many_values(self):
+        for topology in ['FULL', 'ER']:
+            nodes = Network.generate_nodes(n_nodes=5, topology=topology)
+            mempool = Mempool()
+            for node in nodes:
+                node.attach_mempool(mempool)
+
+            for i in range(5):
+                mempool.mine()
+
+            for node in nodes:
+                node.prepare_nomination_msg()
+                node.prepare_nomination_msg()
```

Comment:
double node.prepare_nomination_msg()?

Reply:
unit-tests usually have two major methods: setUp(), tearDown() -> in the tearDown() we clean up cache, in the setUp we set up sut and reset values

Reply:
> double node.prepare_nomination_msg()?

prepare_nomination_msg adds one Value to the state, so this function tests that Values are correctly extended to the state by calling it twice.

Reply:
> unit-tests usually have two major methods: setUp(), tearDown() -> in the tearDown() we clean up cache, in the setUp we set up sut and reset values

Maybe it would be good if we separate the test file into a different number each testing different parts, as some unit tets such as test_initialize_simulator wouldn't use setUp like the rest of the tests. I can do that in a new fix branch